### ISSUE TITLE
Code light?

### ISSUE THREAD

**Issue by zomgugoff** [vipoo/yellow-msx-series-for-rc2014#14]
I can't seem to get the code LED to turn on. I assumed it had something to do with the code key but it doesn't activate the light. I'm not seeing any mention of it in the docs. The key works (Japanese ROM). I had the CODE_LED jumpers installed on the PPI and Game boards. Probing the PPI header, I don't see any activity with or without keypresses. Was this implemented? (Maybe I'm missing the use case?)

**Comment by vipoo** [comment]
This one I am not sure about - i think the CODE is a Japanese thing - so would require the appropriate ROM code.  Not something I have tested.

I think when i original was designing this, I figured out the appropriate OUT statements in basic and confirm it work - but thats now going back a few years for me.  

Sorry I can be more helpful

**Comment by zomgugoff** [comment]
I've found reference in the keyboard [readme](https://github.com/openMSX/openMSX/blob/master/doc/keyboard.readme) for OpenMSX that refers to this key and its behavior. Apparently it's the KANA key and it's supposed to be a toggle. I see a KANA light on photos of real MSX machines also.

**Comment by vipoo** [comment]
Yep, again something i recall doing tested by figureiung out the appropriate PPI config and confirming it works.  But my understanding was that without a Japanese's ROM image it may not have the required software to be driven - again sorry cant be more helpful on this.

**Comment by zomgugoff** [comment]
Your build tools generate C-BIOS roms for different regions, including Japan. I'm using a Japanese ROM. Characters work. It's just this key that's causing issue when it comes to some games.

**Comment by vipoo** [comment]
I see - sorry, this is not something I know very much about.

**Comment by zomgugoff** [comment]
After building an Omega MSX and the associcated ROM, I decided to try the main BIOS from that on the Yellow. The code light now works. I'm not sure what other differences there are in the ROM, but it boots and seems to work. Still having certain programs crash(disk magazines, demos), though.

**Comment by dinoboards** [comment]
Good to know - indicates there is a software issue.  As mentioned before, the MSX BIOS/BASIC are dissasembled reassembled code bases - so very easy for such a regression.  Will need to do some debugging of the code to find it.

Thanks again for the diagnosing

Cheers
